### DISCUSSION TITLE
Support for mTLS

### DISCUSSION THREAD

**Discussion by ryanoolala** [upptime/upptime#632] [Ideas]
Hello,

First of all thanks for building this tool!

I was looking for a way to support a client certificate in the request for sites/APIs that requires mTLS. Is this something that can be done today?

**Comment by AnandChowdhary** [comment]
Unfortunately this is not currently supported but if cURL supports this, then we can easily add it too.

**Reply by ryanoolala** [reply]
Hello,

yes i think it is possible, understand that uptime is using node-libcurl?

i believe the option would be the following https://curl.se/libcurl/c/CURLOPT_SSLCERT.html
to enable the use of a client certificate. Along with this there is also possibly the need to set a few other options as seen at https://curl.se/libcurl/c/simplessl.html
